gentests/books/books_test.go

```diff
@@ -50,34 +50,48 @@ func TestBooks(t *testing.T) {
 }
 
 type BookForm struct {
-	Author  string    `xml:"urn:books author"`
-	Title   string    `xml:"urn:books title"`
-	Genre   string    `xml:"urn:books genre"`
-	Price   float32   `xml:"urn:books price"`
-	Pubdate time.Time `xml:"urn:books pub_date"`
-	Review  string    `xml:"urn:books review"`
-	Name    string    `xml:"name,attr,omitempty"`
+	Author             string     `xml:"urn:books author"`
+	Title              string     `xml:"urn:books title"`
+	Genre              string     `xml:"urn:books genre"`
+	Price              float32    `xml:"urn:books price"`
+	Pubdate            time.Time  `xml:"urn:books pub_date"`
+	Firstrevisiondate  *time.Time `xml:"urn:books first_revision_date,omitempty"`
+	Secondrevisiondate *time.Time `xml:"urn:books second_revision_date,omitempty"`
```

Comment:
With no worthy understanding here, wouldn't it be better to structure revisions as a slice of revision types instead of hardcoding as fields?

Reply:
That would make sense! Except this is just a test of generated code. The BookForm struct and its marshaling and unmarshaling methods are generated using the `//go:generate go run _testgen/testgen.go` command in the `/gentests/gentest.go` file. I can definitely see how that isn't immediately clear though.